app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/BrowserNavigationOverlay.kt

```diff
@@ -266,12 +253,17 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
 
         pocketViewModel.state.observe(fragment.viewLifecycleOwner, Observer { state ->
             when (state) {
-                is PocketViewModel.State.Error -> showMegaTileError()
+                is PocketViewModel.State.Error -> {
+                    pocketVideoMegaTileView.visibility = View.VISIBLE
+                    showMegaTileError()
+                }
                 is PocketViewModel.State.Feed -> {
+                    pocketVideoMegaTileView.visibility = View.VISIBLE
                     pocketVideoMegaTileView.setContent(state.feed)
                 }
+                is PocketViewModel.State.NotDisplayed -> pocketVideoMegaTileView.visibility = View.GONE
                 null -> return@Observer
-            }
+            }.forceExhaustive
```

Comment:
Is `forceExhaustive` necessary if there is an else statement?

Reply:
IMO, `forceExhaustive` should be on every when expression that doesn't return a value (it was a mistake on my part not to have it here in the first place).  Having an else statement will cause an otherwise non-exhaustive when expression to compile, even with `forceExhaustive` on, but for that reason I prefer to avoid else branches when possible.  The advantage here is that, if we ever add a new type to PocketViewModel.State this will force us to update the code everywhere it's used.